shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -209,9 +209,84 @@ static UITextContentType ToUITextContentType(NSArray<NSString*>* hints) {
   return hints[0];
 }
 
-static NSString* uniqueIdFromDictionary(NSDictionary* dictionary) {
+static NSString* autofillIdFromDictionary(NSDictionary* dictionary) {
   NSDictionary* autofill = dictionary[@"autofill"];
-  return autofill == nil ? nil : autofill[@"uniqueIdentifier"];
+  if (autofill) {
+    return autofill[@"uniqueIdentifier"];
+  }
+
+  // When autofill is nil, the field may still need an autofill id
+  // if the field is for password.
+  return [dictionary[@"obscureText"] boolValue] ? @"password" : nil;
+}
+
+// There're 2 types of autofills on native iOS:
+// - Regular autofill, includes contact information autofill and
+//   one-time-code autofill, takes place in the form of predictive
+//   text in the quick type bar. This type of autofill does not save
+//   user input.
+// - Password autofill, includes automatic strong password and regular
+//   password autofill. The former happens automatically when a
+//   "new password" field is detected, and only that password field
+//   will be populated. The latter appears in the quick type bar when
+//   an eligible input field becomes the first responder, and may
+//   fill both the username and the password fields. iOS will attempt
+//   to save user input for both kinds of password fields.
+typedef NS_ENUM(NSInteger, FlutterAutofillType) {
+  // The field does not have autofillable content. Additionally if
+  // the field is currently in the autofill context, it will be
+  // removed from the context without triggering autofill save.
+  FlutterAutofillTypeNone,
+  FlutterAutofillTypeRegular,
+  FlutterAutofillTypePassword,
+};
+
+static BOOL isFieldPasswordRelated(NSDictionary* configuration) {
+  if (@available(iOS 10.0, *)) {
+    BOOL isSecureTextEntry = [configuration[@"obscureText"] boolValue];
+    if (isSecureTextEntry)
+      return YES;
+
+    if (!autofillIdFromDictionary(configuration)) {
+      return NO;
+    }
+    NSDictionary* autofill = configuration[@"autofill"];
```

Comment:
For all these string literals that are shared across the code it would be nice to have const variables for them to avoid errors.
`static NSString *const APIPathFoo = APIBasePath @"foo";`